test/cpp/test_symint.cpp

```diff
@@ -149,21 +149,37 @@ TEST(SymintTest, TestDynamicSymintArithmetic) {
 
   c10::SymInt a =
       c10::make_intrusive<XLASymIntNodeImpl>(size_abs_node)->toSymInt();
-
   c10::SymInt b =
       c10::make_intrusive<XLASymIntNodeImpl>(size_relu_node)->toSymInt();
 
-  c10::SymInt c = a * b;
+  c10::SymInt c = a + b;
+  auto size_add_symnode =
+      dynamic_cast<XLASymIntNodeImpl*>(c.toSymIntNodeImpl().get());
+  ASSERT_TRUE(size_add_symnode);
+  auto size_add =
+      std::dynamic_pointer_cast<torch_xla::SizeAdd>(size_add_symnode->node());
+  ASSERT_EQ(size_add->operands().at(0).node, size_abs_node.get());  //
+  ASSERT_EQ(size_add->operands().at(1).node, size_relu_node.get());
 
+  c = a * b;
   auto size_mul_symnode =
       dynamic_cast<XLASymIntNodeImpl*>(c.toSymIntNodeImpl().get());
   ASSERT_TRUE(size_mul_symnode);
-
   auto size_mul =
       std::dynamic_pointer_cast<torch_xla::SizeMul>(size_mul_symnode->node());
   ASSERT_TRUE(size_mul);
   ASSERT_EQ(size_mul->operands().at(0).node, size_abs_node.get());
   ASSERT_EQ(size_mul->operands().at(1).node, size_relu_node.get());
+
+  c = a / b;
+  auto size_floordiv_symnode =
+      dynamic_cast<XLASymIntNodeImpl*>(c.toSymIntNodeImpl().get());
+  ASSERT_TRUE(size_floordiv_symnode);
+  auto size_floordiv =
+      std::dynamic_pointer_cast<SizeDiv>(size_floordiv_symnode->node());
+  ASSERT_TRUE(size_floordiv);
+  ASSERT_EQ(size_floordiv->operands().at(0).node, size_abs_node.get());
+  ASSERT_EQ(size_floordiv->operands().at(1).node, size_relu_node.get());
```

Comment:
I have another way of testing and I think it might be sightly easier to understand. The difference begins at "// Diff start":
```
TEST(SymintTest, TestDynamicSymintArithmeticAnother) {
  torch::lazy::Value scalar_value =
      torch::lazy::Value(ScalarOp(1.0, xla::F32), 0);

  std::vector<int64_t> target_size = {10, 20, 30};
  torch::lazy::NodePtr expand_node =
      torch::lazy::MakeNode<Expand>(scalar_value, target_size);
  torch::lazy::Value expand_value = torch::lazy::Value(expand_node, 0);

  std::vector<torch::lazy::Shape> abs_lazy_shapes =
      std::vector<torch::lazy::Shape>{
          torch::lazy::Shape(torch::kFloat, {10, 20, 30})};

  std::vector<torch::lazy::Shape> relu_lazy_shapes =
      std::vector<torch::lazy::Shape>{
          torch::lazy::Shape(torch::kFloat, {10, 20, 30})};

  torch::lazy::NodePtr abs_node =
      torch::lazy::MakeNode<Abs>(expand_value, std::move(abs_lazy_shapes));
  torch::lazy::NodePtr relu_node =
      torch::lazy::MakeNode<Relu>(expand_value, std::move(relu_lazy_shapes));

  torch::lazy::NodePtr size_abs_node = torch::lazy::MakeNode<SizeNode>(
      torch::lazy::Value{abs_node, 0}, /*dim=*/0);
  torch::lazy::NodePtr size_relu_node = torch::lazy::MakeNode<SizeNode>(
      torch::lazy::Value{relu_node, 0}, /*dim=*/0);

  // Diff start
  c10::SymIntNode a =
      c10::make_intrusive<XLASymIntNodeImpl>(size_abs_node);
  c10::SymIntNode b =
      c10::make_intrusive<XLASymIntNodeImpl>(size_relu_node);

  auto a_symnode =
      dynamic_cast<XLASymIntNodeImpl*>(a.get());
  ASSERT_TRUE(a_symnode);
  c10::SymIntNode c_symnode = a_symnode->add(b);
  auto c_xla_symnode = dynamic_cast<XLASymIntNodeImpl*>(c_symnode.get());
  auto size_add =
      std::dynamic_pointer_cast<SizeAdd>(c_xla_symnode->node());
  ASSERT_TRUE(size_add);
  ASSERT_EQ(size_add->operands().at(0).node, size_abs_node.get());
  ASSERT_EQ(size_add->operands().at(1).node, size_relu_node.get());
  // Diff ends
}
```
wdyt?

Reply:
They seem pretty similar to me. If all else is equal, I'd land what you already have.